src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -512,7 +512,7 @@ private void getPG(PostgresClient postgresClient, User user, Handler<AsyncResult
    }else{
      Criterion c = new Criterion(
        new Criteria().addField(UserGroupAPI.ID_FIELD_NAME).setJSONB(false).
-       setOperation("=").setValue("'"+pgId+"'"));
+       setOperation("=").setVal(pgId));
      /** check if the patron group exists, if not, can not add the user **/
      postgresClient.get(
        UserGroupAPI.GROUP_TABLE, Usergroup.class, c, true, false, check -> {
```

Comment:
Can you drop Criteria/Criteria and directly use a method that takes the pgId?

      postgresClient.getByIdAsString(UserGroupAPI.GROUP_TABLE, pgId, check -> {
         if (check.succeeded()) {
           if (check.result() == null) {
             handler.handle(io.vertx.core.Future.succeededFuture(0));
           } else {
             handler.handle(io.vertx.core.Future.succeededFuture(1));
           }
         }